File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-2.cpp
class Solution {
public:
    int findMin(vector<int> &nums) {
        if (nums[0] <= nums[nums.size()-1])
        {
            return nums[0];
        }
        int cut = -1;
        for (int i = 1; i < nums.size(); ++i)
        {
            if (nums[i] < nums[i-1])
            {
                cut = i;
            }
        }
        return nums[cut];
    }
};

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-2.py
import numpy as np
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        final_prod = [1]*len(nums)
        for i in range(len(nums)):
            for j in range(len(nums)):
                if j != i:
                   final_prod[i] =  final_prod[i]*nums[j]
        return final_prod
        